lib/src/net.dart

```diff
@@ -1,23 +1,23 @@
 part of rethinkdb;
 
 class Query extends RqlQuery {
-  p.Query_QueryType _type;
-  int _token;
-  RqlQuery _term;
-  Map _globalOptargs;
-  Cursor _cursor;
+  final p.Query_QueryType _type;
+  final int _token;
+  final RqlQuery? _term;
+  final Map? _globalOptargs;
+  Cursor? _cursor;
   final Completer _queryCompleter = Completer();
 
   Query(this._type, this._token, [this._term, this._globalOptargs]);
 
   serialize() {
-    List res = [_type.value];
+    var res = [_type.value];
```

Comment:
Here we would need something like `var res = <dynamic>[_type.value];` (as mentioned in https://dart.dev/guides/language/type-system) to not throw an error in the tests like the one below:

```sh
lib/src/net.dart:24:15: Error: The argument type 'Map<dynamic, dynamic>' can't be assigned to the parameter type 'int'.
   - 'Map' is from 'dart:core'.
        res.add(optargs);
                ^
```